File: Codeforces/Round #639 (Div. 2)/B.cpp
#include<bits/stdc++.h>
#define ll long long 
using namespace std;

int main(){
    ll t, n;
    

    cin >> t;
    while(t--){
        cin >> n;
        
        vector<ll> arr;
        int h=2;
        arr.push_back(2);
        while(arr.back()<n){
            arr.push_back(arr.back()+3*h-1);
            h++;
        }

        ll cnt=0;
        for(int i=arr.size()-1;i>=0;i--){
            while(arr[i]<=n){
                n-=arr[i];
                cnt++;
            }
        }

        cout << cnt << '\n';
    }
    return 0;
}

File: Codeforces/Round #639 (Div. 2)/B.py
arr = [2]
h = 1

t = int(input())
for i in range(t):
    n = int(input())

    while(arr[-1] <= n):
        h+=1
        arr.append(arr[-1]+h*2+h-1)
    
    cnt = 0
    i = len(arr)-1
    while(i>=0):
        if(arr[i]<=n):
            cnt += n//arr[i]
            n = n%arr[i]
        i-=1
    print(cnt)
